src/components/Trade/TradeInput.tsx

```diff
@@ -49,6 +59,11 @@ export const TradeInput = ({ history }: RouterProps) => {
     state: { wallet }
   } = useWallet()
 
+  const cryptoBalance = useAppSelector(state =>
+    selectPortfolioCryptoHumanBalanceByAssetId(state, buyAsset?.currency?.caip19)
```

Comment:
```javascript
  const sellAssetBalance = useAppSelector(state =>
    selectPortfolioCryptoHumanBalanceByAssetId(state, sellAsset?.currency?.caip19)
  )
  const hasValidBalance = bnOrZero(sellAssetBalance).gt(0)
```

That'll do the trick. Checking for valid balance for swap has a few more ramifications to it (checking the number according to the rate and substracting gas for tx) and isn't part of the ACC so it is fine not to tackle it here!

Reply:
Nice catch 🙌 . Yea I was also thinking about including the gas in the comparison, but that is a bit outside of this issue.